tests/python/pants_test/backend/python/rules/test_pex.py

```diff
@@ -64,7 +68,38 @@ def hashify_optional_collection(iterable):
       with zipfile.ZipFile(os.path.join(tmp_dir, "test.pex"), "r") as pex:
         with pex.open("PEX-INFO", "r") as pex_info:
           pex_info_content = pex_info.readline().decode()
-    return json.loads(pex_info_content)
+          pex_list = pex.namelist()
+    return { 'pex': requirements_pex, 'info': json.loads(pex_info_content), 'files': pex_list }
+
+  def create_pex_and_get_pex_info(
+    self, *, requirements=None, entry_point=None, interpreter_constraints=None,
+    input_files: Digest = None, source_dirs = (),
+  ) -> Dict:
+    return self.create_pex_and_get_all_data(requirements=requirements, entry_point=entry_point, interpreter_constraints=interpreter_constraints,
+        input_files=input_files, source_dirs=source_dirs)['info']
+
+  def test_generic_pex_creation(self) -> None:
+    input_files_content = InputFilesContent((
+      FileContent(path='main.py', content=b'print("from main")', is_executable=False),
+      FileContent(path='subdir/sub.py', content=b'print("from sub")', is_executable=False),
+    ))
+
+    input_files, = self.scheduler.product_request(Digest, [input_files_content])
+    entry_point = 'main'
```

Comment:
Nit / subjective style preference. I find this better to inline because you use a kwarg on the next line. If you didn't use the kwarg, then I would agree with having this variable to make it less of a "magic value".